open_event/static/js/admin/event/scheduling.js

```diff
@@ -0,0 +1,512 @@
+    return returnVal;
+}
+
+/**
+ * @return {boolean}
+ */
+function horizontallyBound($parentDiv, $childDiv, offsetAdd) {
+    var pageWidth = $parentDiv.width();
+    var pageHeight = $parentDiv.height();
+    var pageTop = $parentDiv.offset().top;
+    var pageLeft = $parentDiv.offset().left;
+    var elementWidth = $childDiv.width();
+    var elementHeight = $childDiv.height();
+    var elementTop = $childDiv.offset().top;
+    var elementLeft = $childDiv.offset().left;
+    var offset = 50 + offsetAdd;
+    return !!((elementLeft + offset >= pageLeft) && (elementTop + offset >= pageTop) && (elementLeft + elementWidth <= pageLeft + pageWidth) && (elementTop + elementHeight <= pageTop + pageHeight));
+}
+
+function restrictionCheck(x, $sessionElement) {
+    return (horizontallyBound($(".tracks"), $sessionElement, 0));
+}
+
+function overDraggable($sessionElement) {
+    return collision($(".tracks"), $sessionElement);
+}
+
+// Borrowed from http://stackoverflow.com/a/5541252/1562480. Author: BC.
+function collision($div1, $div2) {
+    var x1 = $div1.offset().left;
+    var y1 = $div1.offset().top;
+    var h1 = $div1.outerHeight(true);
+    var w1 = $div1.outerWidth(true);
+    var b1 = y1 + h1;
+    var r1 = x1 + w1;
+    var x2 = $div2.offset().left;
+    var y2 = $div2.offset().top;
+    var h2 = $div2.outerHeight(true);
+    var w2 = $div2.outerWidth(true);
+    var b2 = y2 + h2;
+    var r2 = x2 + w2;
+    return !(b1 < y2 || y1 > b2 || r1 < x2 || x1 > r2);
+}
+
+function updateColor($element) {
+    $element.css("background-color", palette.random("800"));
+    $element.css("background-color", palette.random("800"));
+}
+
+function updateElementTime($element) {
+    var topTime = moment.utc({hour: time.start.hours, minute: time.start.minutes});
+    var mins = pixelsToMinutes($element.outerHeight(false));
+    var topInterval = pixelsToMinutes($element.data("top"), true);
+
+    var startTimeString = topTime.add(topInterval, 'm').format("LT");
+    var endTimeString = topTime.add(mins, "m").format("LT");
+
+    $element.data("start-time", startTimeString).data("end-time", endTimeString);
+    $element.attr("data-original-title", startTimeString + " to " + endTimeString);
+    $element.tooltip("show");
+}
+
+function initializeInteractables() {
+
+    $tracks = $(".track");
+
+    interact(".session")
+        .draggable({
+            // enable inertial throwing
+            inertia: false,
+            // enable autoScroll
+            autoScroll: {
+                container: $(".track-container")[0],
+                margin: 50,
+                distance: 5,
+                interval: 10
+            },
+            restrict: {
+                restriction: ".draggable-holder"
+            },
+            // call this function on every dragmove event
+            onmove: function (event) {
+                var $sessionElement = $(event.target),
+                    x = (parseFloat($sessionElement.data('x')) || 0) + event.dx,
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/triplez/open-event-orga-server/file/2650806054/issues/source?bid=3233864&fileBranchId=3324174#l181)